addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -100,14 +99,14 @@ def portal_my_opportunities(self, page=1, date_begin=None, date_end=None, sortby
         domain = self.get_domain_my_opp(request.env.user)
 
         today = fields.Date.today()
-        this_week_end_date = fields.Date.to_string(fields.Date.from_string(today) + datetime.timedelta(days=7))
+        user_activity_leads = CrmLead.search([('activity_user_id', '=', request.env.user.id)]).ids if filterby == 'no_activities' else []
```

Comment:
Helloo @awa-odoo
I initially did the code with same logic, i.e., directly used `('activity_user_id', '!=', request.env.user.id)` in domain and did or with `"activity_ids", "=", False`. But, it will not work properly when there are multiple activities scheduled of multiple users in a single opportunity, as activity_user_id just brings a single record. 
Link: https://drive.google.com/file/d/1U5w7Pmr25dYi0KHySX_Gq_qyFkDmbT8R/view?usp=drivesdk

Thus, I did it the other way around i.e., first found the opportunities with current user's activity and then not included them in my filter. :sweat_smile:

Reply:
Can we try:

`["|", ("activity_ids", "=", False), ("activity_ids", "not any", [("user_id", "=", request.env.user.id)])]`

Reply:
Helloo @awa-odoo,
Yes it did work. Though, I think we don't need `("activity_ids", "=", False)` separately as `("activity_ids", "not any", [("user_id", "=", request.env.user.id)])` will alone filter out the necessary opportunities..
Please correct me if I have missed something.
Thankss.. :smile: